Angular remote application in standalone doesn't generate environment files

### Current Behavior

When generating an angular remote app with the "standalone" flag, the generated "boostrap.ts" contains an error about not finding the file "environments/environment".

### Expected Behavior

Either the "environments/environment.ts" file is generated or the "apps/cart/src/boostrap.ts" should not use it.

### Github Repo

https://github.com/KevinBonnoron/nx-angular-standalone-remote-bug

### Steps to Reproduce

1. yarn create nx-workspace mf-app --preset=apps --nxCloud=false
2. cd mf-app
3. yarn add -D @nrwl/angular
4. yarn nx g @nrwl/angular:host web --remotes=shop,cart --standalone
5. yarn nx serve web --devRemotes=shop,cart

### Nx Report

```shell
Node : 18.12.1
OS   : linux x64
yarn : 1.22.19

nx : 15.5.2
@nrwl/angular : 15.5.2
@nrwl/cypress : 15.5.2
@nrwl/detox : Not Found
@nrwl/devkit : 15.5.2
@nrwl/esbuild : Not Found
@nrwl/eslint-plugin-nx : 15.5.2
@nrwl/expo : Not Found
@nrwl/express : Not Found
@nrwl/jest : 15.5.2
@nrwl/js : 15.5.2
@nrwl/linter : 15.5.2
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : Not Found
@nrwl/nx-cloud : Not Found
@nrwl/nx-plugin : Not Found
@nrwl/react : Not Found
@nrwl/react-native : Not Found
@nrwl/rollup : Not Found
@nrwl/schematics : Not Found
@nrwl/storybook : Not Found
@nrwl/web : Not Found
@nrwl/webpack : 15.5.2
@nrwl/workspace : 15.5.2
@nrwl/vite : Not Found
typescript : 4.8.4
---------------------------------------
Local workspace plugins:
---------------------------------------
Community plugins:
```


### Failure Logs

```shell
./apps/cart/src/bootstrap.ts:1:0-57 - Error: Module not found: Error: Can't resolve './environments/environment' in '/workspaces/nx-angular15/apps/cart/src'

Error: apps/cart/src/bootstrap.ts:1:29 - error TS2307: Cannot find module './environments/environment' or its corresponding type declarations.

1 import { environment } from './environments/environment';
```


### Additional Information

_No response_

--- a/packages/angular/src/generators/setup-mf/lib/fix-bootstrap.ts
+++ b/packages/angular/src/generators/setup-mf/lib/fix-bootstrap.ts
@@ -28,16 +28,11 @@ export function fixBootstrap(tree: Tree, appRoot: string, options: Schema) {
   );
 }
 
-const standaloneBootstrapCode = `import {environment} from "./environments/environment";
-import {enableProdMode, importProvidersFrom} from "@angular/core";
+const standaloneBootstrapCode = `import {importProvidersFrom} from "@angular/core";
 import {bootstrapApplication} from "@angular/platform-browser";
 import {RouterModule} from "@angular/router";
 import {RemoteEntryComponent} from "./app/remote-entry/entry.component";
 import {appRoutes} from "./app/app.routes";
-
-if (environment.production) {
-  enableProdMode();
-}
 
 bootstrapApplication(RemoteEntryComponent, {
   providers: [

--- a/packages/angular/src/generators/setup-mf/lib/fix-bootstrap.ts
+++ b/packages/angular/src/generators/setup-mf/lib/fix-bootstrap.ts
@@ -1,12 +1,17 @@
 import type { Tree } from '@nrwl/devkit';
 import { joinPathFragments } from '@nrwl/devkit';
 import type { Schema } from '../schema';
+import { getInstalledAngularMajorVersion } from '../../utils/version-utils';
 
 export function fixBootstrap(tree: Tree, appRoot: string, options: Schema) {
   const mainFilePath = joinPathFragments(appRoot, 'src/main.ts');
   const bootstrapCode = tree.read(mainFilePath, 'utf-8');
+  const installedAngularMajor = getInstalledAngularMajorVersion(tree);
   if (options.standalone) {
-    tree.write(`${appRoot}/src/bootstrap.ts`, standaloneBootstrapCode);
+    tree.write(
+      `${appRoot}/src/bootstrap.ts`,
+      standaloneBootstrapCode(installedAngularMajor === 14)
+    );
   } else {
     tree.write(joinPathFragments(appRoot, 'src/bootstrap.ts'), bootstrapCode);
   }
@@ -28,12 +33,23 @@ export function fixBootstrap(tree: Tree, appRoot: string, options: Schema) {
   );
 }
 
-const standaloneBootstrapCode = `import {importProvidersFrom} from "@angular/core";
+const standaloneBootstrapCode = (
+  includeEnvironments: boolean = false
+) => `import {importProvidersFrom} from "@angular/core";
 import {bootstrapApplication} from "@angular/platform-browser";
 import {RouterModule} from "@angular/router";
 import {RemoteEntryComponent} from "./app/remote-entry/entry.component";
 import {appRoutes} from "./app/app.routes";
-
+${
+  includeEnvironments
+    ? `import {enableProdMode} from '@angular/core';
+import {environment} from './environments/environment';
+if(environment.production) {
+  enableProdMode();
+}
+`
+    : ``
+}
 bootstrapApplication(RemoteEntryComponent, {
   providers: [
     importProvidersFrom(

--- a/packages/angular/src/generators/setup-mf/setup-mf.spec.ts
+++ b/packages/angular/src/generators/setup-mf/setup-mf.spec.ts
@@ -317,6 +317,54 @@ describe('Init MF', () => {
     });
   });
 
+  it('should generate bootstrap with environments for ng14', async () => {
+    // ARRANGE
+    updateJson(tree, 'package.json', (json) => ({
+      ...json,
+      dependencies: {
+        ...json.dependencies,
+        '@angular/core': '14.1.0',
+      },
+    }));
+
+    await applicationGenerator(tree, {
+      name: 'ng14',
+      routing: true,
+      standalone: true,
+    });
+
+    // ACT
+    await setupMf(tree, {
+      appName: 'ng14',
+      mfType: 'host',
+      routing: true,
+      standalone: true,
+    });
+
+    // ASSERT
+    expect(tree.read('apps/ng14/src/bootstrap.ts', 'utf-8'))
+      .toMatchInlineSnapshot(`
+      "import {importProvidersFrom} from \\"@angular/core\\";
+      import {bootstrapApplication} from \\"@angular/platform-browser\\";
+      import {RouterModule} from \\"@angular/router\\";
+      import {RemoteEntryComponent} from \\"./app/remote-entry/entry.component\\";
+      import {appRoutes} from \\"./app/app.routes\\";
+      import {enableProdMode} from '@angular/core';
+      import {environment} from './environments/environment';
+      if(environment.production) {
+        enableProdMode();
+      }
+
+      bootstrapApplication(RemoteEntryComponent, {
+        providers: [
+          importProvidersFrom(
+            RouterModule.forRoot(appRoutes, {initialNavigation: 'enabledBlocking'})
+          )
+        ]
+      });"
+    `);
+  });
+
   it('should add a remote to dynamic host correctly', async () => {
     // ARRANGE
     await setupMf(tree, {

--- a/packages/angular/src/utils/nx-devkit/testing.ts
+++ b/packages/angular/src/utils/nx-devkit/testing.ts
@@ -53,15 +53,9 @@ export function createApp(
   tree.write(
     `/apps/${appName}/src/main.ts`,
     `
-    import { enableProdMode } from '@angular/core';
     import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 
     import { AppModule } from './app/app.module';
-    import { environment } from './environments/environment';
-
-    if (environment.production) {
-      enableProdMode();
-    }
 
     platformBrowserDynamic()
       .bootstrapModule(AppModule)

